client/packages/electron/src/keyboardScanner/keyboardScanner.ts

```diff
@@ -0,0 +1,231 @@
+import { BrowserWindow } from 'electron';
+import { IPC_MESSAGES } from '../shared';
+import { BarcodeScanner } from '@common/hooks';
+
+// Some modifier keys are passed through (no preventDefault)
+const keysToPassThrough = [
+  'Backspace',
+  'Tab',
+  'Enter',
+  'Shift',
+  'Control',
+  'Alt',
+  'Pause',
+  'CapsLock',
+  'Escape',
+  'Space',
+  'PageUp',
+  'PageDown',
+  'End',
+  'Home',
+  'ArrowLeft',
+  'ArrowUp',
+  'ArrowRight',
+  'ArrowDown',
+  'PrintScreen',
+  'Insert',
+  'Delete',
+  'Clear',
+];
+
+// Scan Log
+const PROCESSOR_INTERVAL = 50;
+
+type KeyEvent = { input: Electron.Input; ms: number };
+
+// Some keys need to be removed (mention about the thig)
+const ignoreKeys = (input: Electron.Input) =>
+  // Just focusing on keyDown events
+  input.type != 'keyDown' ||
+  // When input holding key press
+  input.isAutoRepeat;
+
+type KeyboardScannerSettings = {
+  maxMSBetweenScannerKeyInputs: number;
+  minBarcodeLength: number;
+};
+
+const KEYBOARD_SCANNER: BarcodeScanner = {
+  vendorId: 1,
+  productId: 1,
+  connected: true,
+};
+
+export class KeyboardScanner {
+  window: BrowserWindow;
+  interval: NodeJS.Timer | undefined;
+  // Buffer can be written concurrent in event handles and when processing input
+  // this lock is to make sure it's edited syncrhonously
+  lockBuffer: boolean;
+  // In scanning mode, when non barcode data entry is made, should allow it (no preventDefault())
+  allowEvents: boolean;
+  scanning: boolean;
+  buffer: KeyEvent[];
+  settings: KeyboardScannerSettings;
+
+  constructor(window: BrowserWindow) {
+    // this.device = this.findDevice();
+    this.window = window;
+    this.scanning = false;
+    this.lockBuffer = false;
+    this.allowEvents = false;
+    this.settings = {
```

Comment:
I thought this was going to be a configurable setting, but decided not to add these configurations yet, but kept the data structure